src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -3891,6 +3891,22 @@
       </Field>
     </Choice>
   </Node>
+  <Node Name="DataPropertyDeclarationSyntax" Base="MemberDeclarationSyntax">
+    <Kind Name="DataPropertyDeclaration" />
+    <Field Name="AttributeLists" Type="SyntaxList&lt;AttributeListSyntax&gt;" Override="true"/>
+    <Field Name="Modifiers" Type="SyntaxList&lt;SyntaxToken&gt;" Override="true"/>
+    <Field Name="DataKeyword" Type="SyntaxToken">
```

Comment:
I understand that we have a principle that our grammar should result in non-overlapping productions. But I wonder if we could achieve that without hard-coding the position of the `data` keyword.

I guess I'm proposing that we treat `data` as one of the modifiers (and so could be re-ordered). That seems doable from a parser perspective, but I'm not sure if we could make that work from a language perspective.

Not blocking this PR or 16.7p3, but may be worth further discussion.

Reply:
I'm not opposed to it, but I found it somewhat easier to think about as closer to the "event" syntax than a modifier, because if we ever did allow another modifier in the declaration I'm not sure I would like the potential reordering.

Reply:
Added a note to test plan

---
In reply to: [436428579](https://github.com/dotnet/roslyn/pull/44883#discussion_r436428579) [](ancestors = 436428579)